src/main/java/org/prebid/server/auction/ExchangeService.java

```diff
@@ -871,6 +901,14 @@ private static void populateWinningBids(TargetingKeywordsCreator keywordsCreator
         }
     }
 
+    private static Bid modifyBid(Bid bid, Boolean returnCreative) {
```

Comment:
Move the logic of this method to `toBid(...)`.
Just add 2 arguments: `boolean returnCreativeBids, boolean returnCreativeVideoBids` and pass they from `toBidResponse(...)` from `cacheInfo` object.